source/fem-dem/gls_sharp_navier_stokes.cc

```diff
@@ -280,50 +286,67 @@ GLSSharpNavierStokesSolver<dim>::generate_cut_cells_map()
                                     particle_id_in_which_this_cell_is_embedded};
         }
     }
+  dof_over_constraint = local_dof_over_constraint;
+  dof_over_constraint.compress(VectorOperation::insert);
 
   // We loop over every fluid cell, and if all of its DOFs are cut we change the
   // status of the cell itself.
   for (const auto &cell : cell_iterator)
     {
       if (cell->is_locally_owned() || cell->is_ghost())
         {
-          overconstrained_fluid_cell_map[cell] = {false, -1};
-
-          bool cell_is_cut;
-          bool cell_is_inside;
-          cell->get_dof_indices(local_dof_indices);
-          std::tie(cell_is_cut, std::ignore, std::ignore) = cut_cells_map[cell];
-          std::tie(cell_is_inside, std::ignore) = cells_inside_map[cell];
-          if (!cell_is_cut && !cell_is_inside)
+          overconstrained_fluid_cell_map[cell] = {false, -1, -1};
```

Comment:
Ideally, we would fill the map only if the cell is overconstrained to avoid huge maps of cells. The same modification could be applied on the cut cell map maybe we should keep this for an other PR only on this small correction.